frontend/src/metabase/query_builder/components/DimensionList.jsx

```diff
@@ -60,9 +61,14 @@ export default class DimensionList extends Component {
 
   itemIsSelected = item => {
     const dimensions = this._getDimensions();
+    const { dimension } = item;
     return (
       item.dimension &&
-      _.any(dimensions, d => item.dimension.isSameBaseDimension(d))
+      _.any(dimensions, d => {
+        return isVirtualCardId(dimension?.query()?.table()?.id)
+          ? d.isSameBaseDimension(dimension.withoutJoinAlias())
+          : d.isSameBaseDimension(dimension);
```

Comment:
Seems worthwhile to have a comment in the code along the lines of the PR comment.

Silly question: just using `dimension.withoutJoinAlias()` for everything wouldn't work here, would it?

Reply:
Good call out! This apparently works and is much simpler 🥇